tests/DCPS/XTypes/Pub/XTypesPublisher.cpp

```diff
@@ -409,11 +454,51 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
     }
   }
 
-  ACE_OS::sleep(ACE_Time_Value(3, 0));
+  ACE_DEBUG((LM_DEBUG,"Writer waiting for ack at %T\n"));
+
+  ::ControlStructSeq control_data;
+  ReturnCode_t control_ret = read_i(control_dr, control_pdr, control_data);
+  if (control_ret != RETCODE_OK) {
+    ACE_ERROR((LM_ERROR, "ERROR: control read returned %C\n",
+      OpenDDS::DCPS::retcode_to_string(ret)));
+    return 1;
+  }
+
+  ACE_DEBUG((LM_DEBUG,"Writer sending echo at %T\n"));
+
+  ControlStruct cs;
+  control_ret = control_typed_dw->write(cs, HANDLE_NIL);
+  if (control_ret != RETCODE_OK) {
+    ACE_ERROR((LM_ERROR, "ERROR: control write returned %C\n",
+      OpenDDS::DCPS::retcode_to_string(control_ret)));
+    return 1;
+  }
+
+  condition = control_dw->get_statuscondition();
+  condition->set_enabled_statuses(DDS::PUBLICATION_MATCHED_STATUS);
+
+  ws->attach_condition(condition);
+  DDS::Duration_t p = { 5, 0 };
+  DDS::PublicationMatchedStatus pms;
+  control_dw->get_publication_matched_status(pms);
+  ACE_DEBUG((LM_DEBUG, "count of subs = %d change count = %d at %T\n", pms.current_count, pms.current_count_change));
+  for (int retries = 3; retries > 0 && pms.current_count > 0; retries-- ) {
+    conditions.length(0);
+    control_ret = ws->wait(conditions, p);
```

Comment:
Do we want to log an warning or maybe an error if it times out after every retry?

Reply:
No, the debug comments were there for my investigation. I'll pull the debug out of the loop and report a "failure" afterwards. Since this point is not material to the success of the test there's no need for special processing. The only way for this to fail would be if the peer failed and the infrastructure already deals with that case.

Reply:
I found an inconsistency between the testing of control_ret and the subsequent reporting of ret. So I remedied this by removing the redundant  control_ret value and just use the now non-const value, ret.